memberships/views.py

```diff
@@ -141,3 +145,24 @@ def stripe_webhook(request):
         return stripe_webhook.handle(event)
     except ValueError as e:
         return HttpResponse("Failed to parse stripe payload", status=400)
+
+
+@login_required()
+def details_view(request):
+    pass
+
+
+@login_required()
+def settings_view(request):
+    if not request.method == "POST":
+        return render(
+            request, "memberships/member_settings.html", {"form": MemberSettingsForm(instance=request.user.member)}
+        )
+
+    form = MemberSettingsForm(request.POST, instance=request.user.member)
+    if not form.is_valid():
+        return render(
+            request, "memberships/member_settings.html", {"form": form})
+
+    form.save()
```

Comment:
Can a model `save` method fail? If it can, is there a graceful way we can handle it?

Reply:
the `if not form.is_valid():` ensures that the form is good before we get there, right?

Reply:
It does, but since `save` is writing to the database we could have issues when connecting to the database. The most common cause would be the database is down, but there could be others that I can't think of right now.

We could just let it fail and throw a generic error message page back to the user, but it might also be helpful to provide them with a message stating that we were unable to update their details and to try again later.

I guess this is probably something we should deal with later as there will be lots of places where we write to the database and we would want a consistent way of handling any errors that come from that.

Reply:
Agreed, that's an improvement rather than critical functionality.